src/receive_imf.rs

```diff
@@ -677,12 +677,22 @@ pub(crate) async fn receive_imf_inner(
         let res = if mime_parser.incoming {
             handle_securejoin_handshake(context, &mut mime_parser, from_id)
                 .await
-                .context("error in Secure-Join message handling")?
+                .with_context(|| {
+                    format!(
+                        "error in Secure-Join '{}' message handling",
```

Comment:
Since you are changing the error messages anyway:
```suggestion
                        "Error in Secure-Join '{}' message handling",
```
(https://github.com/chatmail/core/blob/main/STYLE.md#errors)